Fixes #5453 updates cffi & argon2_cffi

kushaldas opened the pull request:
## Status

Ready for review 

## Description of Changes

Fixes #5453

Updates both the releases to the latest, these versions builds
properly on Python3.8 on Ubuntu Focal 20.04

## Testing

- [x] CI is green
- [x] Verify hashes introduced in the PR (for the source tar.gz files with the following)
- [x] https://github.com/freedomofpress/securedrop-debian-packaging/wiki/cffi-1.11.5-to-1.14.2
- [x] https://github.com/freedomofpress/securedrop-debian-packaging/wiki/argon2_cffi-18.1.0-to-20.1.0

## Deployment

Any special considerations for deployment? Consider both:

1. Upgrading existing production instances.
2. New installs.

## Checklist

### If you made changes to the server application code:

- [ ] Linting (`make lint`) and tests (`make test`) pass in the development container

### If you made changes to `securedrop-admin`:

- [ ] Linting and tests (`make -C admin test`) pass in the admin development container

### If you made changes to the system configuration:

- [ ] [Configuration tests](https://docs.securedrop.org/en/latest/development/testing_configuration_tests.html) pass

### If you made non-trivial code changes:

- [ ] I have written a test plan and validated it for this PR

### If you made changes to documentation:

- [ ] Doc linting (`make docs-lint`) passed locally

### If you added or updated a code dependency:

Choose one of the following:

- [x] I have performed a diff review and pasted the contents to [the packaging wiki](https://github.com/freedomofpress/securedrop-debian-packaging/wiki)
- [ ] I would like someone else to do the diff review

kushaldas committed:
Fixes #5453 updates cffi & argon2_cffi

Updates both the releases to the latest, these versions builds
properly on Python3.8 on Ubuntu Focal 20.04

emkll commented on securedrop/requirements/python3/securedrop-app-code-requirements.txt:
```diff
@@ -38,40 +31,35 @@ babel==2.5.1 \
-    --hash=sha256:edabd457cd23a02965166026fd9bfd196f4324fe6032e866d0f3bd0301cd486f \
-    --hash=sha256:fdf1c1dc5bafc32bc5d08b054f94d659422b05aba244d6be4ddc1c72d9aa70fb \
-    # via argon2-cffi, cryptography
+cffi==1.14.0 \
```
cffi version reviewed in the PR description is 1.14.2, as such, the hashes are different. Should we use 1.14.2 here?

kushaldas replied:
I will update it in some time.

emkll reviewed (commented):
Went through the test plan and everything looks good to me: 

- [x] CI is green
- [x] Verify hashes introduced in the PR (for the source tar.gz files with the following)
- [x] https://github.com/freedomofpress/securedrop-debian-packaging/wiki/cffi-1.11.5-to-1.14.2
- [x] https://github.com/freedomofpress/securedrop-debian-packaging/wiki/argon2_cffi-18.1.0-to-20.1.0

One last question for you @kushaldas : would it make sense to track the same version of cffi (1.4.2) in develop-requirements.txt as well? It's currently on 1.4.0 [1](which may have been the source of the initial confusion). It seems fine for me to merge as-is, but curious what you think.

[1] https://github.com/freedomofpress/securedrop/blob/upgrade_ffi_twins/securedrop/requirements/python3/develop-requirements.txt#L109

kushaldas committed:
Updates argon2_cffi and cffi versions

Based on the review feedback we now have the same version in
development and main application requirements files.

emkll reviewed (approved):
Thanks @kushaldas , good to merge when CI passes
